sdk/monitor/opentelemetry-exporter/src/export/trace.ts

```diff
@@ -0,0 +1,27 @@
+import { ExportResult } from "@opentelemetry/core";
+import { ReadableSpan, SpanExporter } from "@opentelemetry/tracing";
+import { readableSpanToEnvelope } from "../utils/spanUtils";
+import { DEFAULT_EXPORTER_CONFIG, AzureExporterConfig } from "../config";
+import { AzureMonitorBaseExporter } from "./exporter";
+
+export class AzureMonitorTraceExporter extends AzureMonitorBaseExporter implements SpanExporter {
+  constructor(options: Partial<AzureExporterConfig> = {}) {
+    super({
+      ...DEFAULT_EXPORTER_CONFIG,
+      ...options,
+    });
+  }
+
+  export(spans: ReadableSpan[], resultCallback: (result: ExportResult) => void): void {
+    this._logger.info(`Exporting ${spans.length} span(s). Converting to envelopes...`);
+    const envelopes = spans.map((span) =>
+      readableSpanToEnvelope(span, this._options.instrumentationKey!, this._logger),
```

Comment:
is there a way we can get out of having to assert instrumentationKey exists here? Needing to force the compiler into ignoring something possibly being undefined feels like a good way to get a bug someday

Reply:
I need to put more thought into this since either a `connectionString` or `instrumentationKey` should be provided, but a `connectionString` will change `instrumentationKey` at runtime. For now, I've made `instrumentationKey` required, but end-users provide a `Partial<>` config